### ISSUE TITLE
[BUG] Float gets autoconverted into integer in query in Peewee 3.1

### ISSUE THREAD

**Issue by Egor3f** [coleifer/peewee#1552]
I have exaclty the same issue, as there: https://github.com/coleifer/peewee/issues/1150
I need to multiply IntegerField by float value in select query (to apply commission or discount to price field).
But in actual SQL query float gets converted to integer, so 0.9 becomes 0.

In peewee v2 I could use `Item.select((Item.price * Passthrough(multiplier)).alias('final_price'))`

There is no Passthrough function in peewee v3.

My database is Mariadb through pymysql, if it matters.

It seeems strange that such simple action is impossible in peewee. And more strange, that in Peewee v2 I could do that, and in new version I can't.

P.S. I don't need any CASTing, just need to pass right value without converting.

**Comment by coleifer** [comment]
Just a thought -- if you have a price and are multiplying it by a float, that could cause issues... you might have better luck using a `DecimalField` which will not introduce precision errors.

I'll look into the issue as reported, just thought I'd mention that as a suggestion.

**Comment by coleifer** [comment]
I just wrote a small test-case and it is passing:

```python
class Item(TestModel):
    price = IntegerField()
    multiplier = FloatField(default=1.)


class TestFieldValueHandling(ModelTestCase):
    requires = [Item]

    def test_int_float_multi(self):
        i = Item.create(price=10, multiplier=0.75)
        i_db = (Item
                .select(Item, (Item.price * Item.multiplier).alias('total'))
                .where(Item.id == i.id)
                .get())
        self.assertEqual(i_db.price, 10)
```

Can you help me understand what I've got wrong in my setup so I can replicate this?

**Comment by coleifer** [comment]
Ahh, got it to fail, nevermind:

```python
        i_db = (Item
                .select(Item, (Item.price * 0.75).alias('total'))
                .where(Item.id == i.id)
                .get())
        self.assertEqual(i_db.price, 10)
        self.assertEqual(i_db.multiplier, .75)
        self.assertEqual(i_db.total, 7.5)  # FAILS -- 0
```

**Comment by coleifer** [comment]
Fixed in: 969c208f34f159b742e81be722e6a409400843ad

You should now be able to pass "False" as the 2nd parameter of `Value` (the 3.0 replacement for Passthrough):

```python
Item.select((Item.price * Value(multiplier, False)).alias('final_price'))
```
